ouroboros-network/src/Ouroboros/Network/ChainFragment.hs

```diff
@@ -379,6 +380,16 @@ dropWhileNewest p c@(c' :> b)
                   | p b       = dropWhileNewest p c'
                   | otherwise = c
 
+takeWhileOldest :: HasHeader block
```

Comment:
Could you add the following property test to `Test.ChainFragment`?

```haskell

prop_takeWhileOldest :: (Block -> Bool) -> TestBlockChainFragment -> Bool
prop_takeWhileOldest p (TestBlockChainFragment chain) =
    CF.takeWhileOldest p chain
 == (CF.fromOldestFirst . L.takeWhile p . CF.toOldestFirst) chain
```

Reply:
Done